dos/src/main.rs

```diff
@@ -414,34 +418,43 @@ fn run_dos_transactions<T: 'static + BenchTpsClient + Send + Sync>(
 
     let mut transaction_generator = TransactionGenerator::new(transaction_params);
 
+    let connection_cache_stats = Arc::new(ConnectionCacheStats::default());
+    let udp_client = UdpTpuConnection::new(target, connection_cache_stats);
+
     let mut count = 0;
     let mut total_count = 0;
     let mut error_count = 0;
     let mut last_log = Instant::now();
+
+    let mut data = Vec::<Vec<u8>>::with_capacity(SEND_BATCH_MAX_SIZE);
```

Comment:
For my own education: what are the gains of this approach over creating a new empty vec over every iteration?

Reply:
It is to reduce the number of memory allocations. But if we use `_async` than it will not work, but async gives more gain than this